crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -228,6 +229,24 @@ def cancel
         }
       }, status: cancel_upgrade[:status]
     end
+  rescue Crowbar::Error::UpgradeCancelError => e
```

Comment:
Why did you move this from model to controller? There are still some errors returned via `cancel_upgrade` and some are handled as exceptions. Why not keep it consistent?